File: app/src/main/java/work/beltran/discogsbrowser/app/common/RecordViewHolder.java
package work.beltran.discogsbrowser.app.common;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import butterknife.BindView;
import butterknife.ButterKnife;
import work.beltran.discogsbrowser.R;
import work.beltran.discogsbrowser.app.release.ReleaseActivity;

/**
 * Created by Miquel Beltran on 10/1/16
 * More on http://beltran.work
 */
class RecordViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
    private Picasso picasso;
    @BindView(R.id.text_title)
    TextView textTitle;
    @BindView(R.id.image_thumb)
    ImageView iamgeThumb;
    @BindView(R.id.text_artist)
    TextView textArtist;
    @BindView(R.id.text_year)
    TextView textYear;
    @BindView(R.id.text_format)
    TextView textFormat;
    private RecordAdapterItem record;

    RecordViewHolder(View itemView, Picasso picasso) {
        super(itemView);
        this.picasso = picasso;
        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(this);
    }

    void bind(RecordAdapterItem record) {
        textTitle.setText(record.getTitle());
        textArtist.setText(record.getArtist());
        textFormat.setText(record.getFormat());
        textYear.setText(record.getYear());
        picasso.load(record.getThumb())
                .fit()
                .centerCrop()
                .into(iamgeThumb);
        this.record = record;
    }

    @Override
    public void onClick(View v) {
        v.getContext().startActivity(ReleaseActivity.createReleaseActivity(v.getContext(), record));
    }
}

File: app/src/main/java/work/beltran/discogsbrowser/app/common/RecordViewHolder.kt
package work.beltran.discogsbrowser.app.common

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView

import com.squareup.picasso.Picasso

import kotlinx.android.synthetic.main.card_record.view.*
import work.beltran.discogsbrowser.R
import work.beltran.discogsbrowser.app.release.ReleaseActivity

/**
 * Created by Miquel Beltran on 10/1/16
 * More on http://beltran.work
 */
class RecordViewHolder(itemView: View)
    : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    private val textTitle: TextView by lazy { itemView.text_title }
    private val thumbImage: ImageView by lazy { itemView.image_thumb }
    private var record: RecordItem? = null

    init {
        itemView.setOnClickListener(this)
    }

    internal fun bind(record: RecordItem) {
        textTitle.text = record.title
        showAlbumArt(record)
        this.record = record
    }

    private fun showAlbumArt(record: RecordItem) {
        if (record.thumb.isEmpty()) {
            thumbImage.setImageResource(R.drawable.music_record)
        } else {
            Picasso.with(thumbImage.context)
                    .load(record.thumb)
                    .placeholder(R.drawable.music_record)
                    .centerCrop()
                    .fit()
                    .into(thumbImage)
        }
    }

    override fun onClick(v: View) {
        v.context.startActivity(ReleaseActivity.createReleaseActivity(v.context, record))
    }
}
